geopandas/geodataframe.py

```diff
@@ -252,9 +252,17 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
 
         Parameters
         ----------
-        col : column label or array
+        col : column label or array-like
+            An existing column name or values to set as the new geometry column.
+            If values (array-like, Series) are passed, then if they are named (Series)
+              the new geometry column will have the corresponding name, otherwise
+              the existing geometry column will be replaced. If there is no existing
+              geometry column, the new geometry column will use the default name
+              "geometry".
         drop : boolean, default False
-            Delete column to be used as the new geometry
+            When specifying a named Series or an existing column name for `col`,
+            controls if the previous geometry column should be dropped from the
+            result. The default of False keeps both the old and new geometry column.
         inplace : boolean, default False
             Modify the GeoDataFrame in place (do not create a new object)
```

Comment:
Sidenote, and not for this PR, but we should also deprecate this keyword at some point (given that pandas is deprecating it generally)
(and for the default case of `inplace=false` we should do `self.copy(deep=False)` instead of a hard copy for pandas>=3)